packages/@aws-cdk/aws-logs/lib/log-retention-provider/index.ts

```diff
@@ -51,6 +51,35 @@ async function createLogGroupSafe(logGroupName: string, region?: string, options
   } while (true); // exit happens on retry count check
 }
 
+//delete a log group
+async function deleteLogGroup(logGroupName: string, region?: string, options?: SdkRetryOptions) {
+  let retryCount = options?.maxRetries == undefined ? 10 : options.maxRetries;
+  const delay = options?.retryOptions?.base == undefined ? 10 : options.retryOptions.base;
+  do {
+    try {
+      const cloudwatchlogs = new AWS.CloudWatchLogs({ apiVersion: '2014-03-28', region, ...options });
+      await cloudwatchlogs.deleteLogGroup({ logGroupName }).promise();
+      return;
+    } catch (error) {
+      if (error.code === 'ResourceNotFoundException') {
```

Comment:
Thanks for your comment! If it doesn't fall silently and someone manually deletes the log group, it will throw an error. How to distinguish this situation and other reasons? Could you please give me some suggestions?

Reply:
I suppose since this would cause the stack deletion to fail we do want this in here.

Reply:
Do you mean that we should keep the code?